uint8 port
int32 position
